staging/src/k8s.io/kubectl/pkg/cmd/create/create_role.go

```diff
@@ -329,6 +330,9 @@ func (o *CreateRoleOptions) RunCreateRole() error {
 		TypeMeta: metav1.TypeMeta{APIVersion: rbacv1.SchemeGroupVersion.String(), Kind: "Role"},
 	}
 	role.Name = o.Name
+	if o.NamespaceOverwritten {
+		role.Namespace = o.Namespace
```

Comment:
I don't even think this condition is necessary, we should just ensure the `role.Namespace` is set.

Reply:
That is what I did originally, but it resulted in `metadata.namespace` being explicitly set to `default` in the YAML output when the user did not use the `--namespace` option. For example:

```bash
kubectl create role foo --verb=get --resource=pod --output=yaml --dry-run
```

would now output this:

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  creationTimestamp: null
  name: foo
  namespace: default
rules:
- apiGroups:
  - ""
  resources:
  - pods
  verbs:
  - get
```

while it used to output this:

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  creationTimestamp: null
  name: foo
rules:
- apiGroups:
  - ""
  resources:
  - pods
  verbs:
  - get
```

Reply:
Fair, please use `EnforceNamespace` as the var name, in that case.

Reply:
Done.